### ISSUE TITLE
Support reading builtins from `[tool.pyright]` in `pyproject.toml`

### ISSUE THREAD

**Issue by C0rn3j** [microsoft/pyright#9471]
Builtins are currently handled via `__builtins__.pyi`, as per the [documentation](https://github.com/microsoft/pyright/blob/main/docs/builtins.md#extending-builtins)

Pyright [supports pyproject.toml](https://github.com/microsoft/pyright/blob/main/docs/configuration.md) for its settings, but builtins are missing from this.

I would welcome if Pyright also enabled defining them through `pyproject.toml`, like the other settings, and like other tools allow.

Here's how Ruff does it, as per their [documentation](https://docs.astral.sh/ruff/settings/#builtins):

```toml
[tool.ruff]
	builtins = ["_"]
```

The existing documentation is also not very clear as to how to define them through the stub file.

EDIT: Managed to do what I wanted via setting the file content to:  

```python
def _(text: str) -> str: ...
# Alternate global without type:
#_ = ...
```

**Comment by erictraut** [comment]
It's not sufficient to specify only the names of the builtins. That might suffice for a linter, but a type checker needs to know the _types_ of the builtin symbols. That requires type declarations and therefore requires a ".pyi" file.

**Comment by C0rn3j** [comment]
I see, in that case it would still be good to improve the current documentation page to add an example.
